src/kirby/components/modal/modal.component.tns.ts

```diff
@@ -0,0 +1,117 @@
+import { Color } from 'tns-core-modules/color';
+import {
+  Component,
+  ViewContainerRef,
+  ComponentFactoryResolver,
+  ViewChild,
+  OnInit,
+  ElementRef,
+  Renderer2,
+} from '@angular/core';
+import { ModalDialogParams } from 'nativescript-angular';
+import { ContentView, ShownModallyData } from 'tns-core-modules/ui/content-view';
+import { StackLayout } from 'tns-core-modules/ui/layouts/stack-layout/stack-layout';
+import { ModalStack } from 'nativescript-windowed-modal';
+
+import { ModalConfig } from './config/modal-config';
+import { ModalCloserService } from './services/modal-closer-service';
+import { ModalConfigHelper } from './helpers/modal-config-helper';
+import { ModalNestedComponentHelper } from './helpers/modal-nested-component-helper';
+
+const style: any = require('sass-extract-loader!./modal.component.scss');
+
+@Component({
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss'],
+})
+export class ModalComponent extends ContentView implements OnInit {
+  @ViewChild('container') viewContainer: ElementRef;
+  config: ModalConfig;
+  modalStack: ModalStack;
+  uid: number;
+
+  constructor(
+    private modalCloserService: ModalCloserService,
+    private params: ModalDialogParams,
+    private modalNestedComponentHelper: ModalNestedComponentHelper,
+    private componentFactoryResolver: ComponentFactoryResolver,
+    public vcRef: ViewContainerRef,
+    private renderer: Renderer2
+  ) {
+    super();
+    this.config = ModalConfigHelper.processOptionalValues(params.context);
+    this.uid = params.context.uid;
+    this.modalCloserService.registerModalCloseRef(this.uid, params.closeCallback);
+  }
+
+  ngOnInit() {
+    this.modalNestedComponentHelper.appendComponent(
+      this.viewContainer,
+      this.vcRef,
+      this.renderer,
+      this.componentFactoryResolver,
+      this.config.component,
+      this.config.uid
+    );
+  }
+
+  showModally(args: ShownModallyData): void {
+    this.modalStack = <ModalStack>args.object;
+    const stackLayout = <StackLayout>args.object;
+    stackLayout.backgroundColor = new Color(0, 0, 0, 0);
+    this.animateModal(stackLayout);
+  }
+
+  // TODO: better to call the modal-service somehow, circular dependency prevents this atm
+  dismissModal(): void {
+    this.params.closeCallback();
```

Comment:
Should be accessible by injecting `Injector` (to a private field) then calling `this.injector.get(ModalService)`

Reply:
Unfortunately this raises multiple warnings regarding circular dependencies, I will leave it as it is